### ISSUE TITLE
multilib policy is not enforced enough

### ISSUE THREAD

**Issue by ignatenkobrain** [openSUSE/libsolv#149]
```
repo system 0 empty
repo available 0 testtags <inline>
#>=Pkg: foo 2 1 x86_64
#>=Pkg: foo 1 1 i686
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job install pkg foo-2-1.x86_64@available
job install pkg foo-1-1.i686@available
result transaction,problems
```

results into:
```
Transaction summary:

2 installed packages:
  - foo-1-1.i686
  - foo-2-1.x86_64
```

From https://bugzilla.redhat.com/show_bug.cgi?id=1326157

**Comment by mlschroe** [comment]
Yes. Do you really want to enforce it for explicit user requests as well?

**Comment by ignatenkobrain** [comment]
@jsilhan, what do you think?

**Comment by mlschroe** [comment]
Btw, you can "fix" that in hawkey/libdnf with SOLVER_NOAUTOSET, so that the solver doesn't assume the arch is intentional for that job.

If SOLVER_NOAUTOSET is used, libsolv does not attempt to guess the intention from the job. I.e. it does not assume SOLVER_SETNAME for SOLVER_SOLVABLE_NAME installs,  SOLVER_SETNAME | SOLVER_SETARCH | SOLVER_SETVENDOR | SOLVER_SETREPO | SOLVER_SETEVR for SOLVER_SOLVABLE installs, and so on. See jobtodisablelist() in solver.c

**Comment by ignatenkobrain** [comment]
that seems like workaround... Would be nice to get flag which would completely enforce to be all multilib arches in sync

**Comment by mlschroe** [comment]
Well, I won't add it if it will not be used, and your question above is not answered yet...

**Comment by jsilhan** [comment]
Currently DNF do queries on packages, find package ids based on filters and then put it into selector to resolve one of given packages. If the flag would handle `multilib_policy=best/all` by itself that would be an improvement. I would especially invite this flag during VERIFY job so it would report these multilib violation when the new flag will be set (if it;s not libsolv doing already).

**Comment by dmach** [comment]
We want the packages to be at the same version.
The system is better auditable and users know exactly what's installed.
Having a libsolv feature (a flag?) for that would be much nicer than hacking this in libdnf/dnf directly.

**Comment by ignatenkobrain** [comment]
@mlschroe friendly ping, we want to lock-in 32bit and 64bit versions of same package to same evr in Fedora.

**Comment by ignatenkobrain** [comment]
(by having solverflag or poolflag)

**Comment by ignatenkobrain** [comment]
> Yes. Do you really want to enforce it for explicit user requests as well?

I think this is still not answered… Yes, this should make sure that multilib packages are kept in sync all the time.

**Comment by mlschroe** [comment]
I can easily add a flag that makes the solver not disable "infarch" rules, but that means that there will be no way to install that foo-1-1.i686 package with dnf (as there is no corresponding foo-1.1.x86_64 package).

**Comment by ignatenkobrain** [comment]
@mlschroe I'll check the YUM behavior and write it back here.

**Comment by dmnks** [comment]
I promised to chime in with some information on how yum handles this.

In yum, we have the following config option:
```
protected_multilib

Either `1' or `0'. This tells yum whether or not it should perform a check to make sure that multilib packages are the same version. For example, if this
option  is  off  (rpm  behavior)  then  in  some  cases it might be possible for pkgA-1.x86_64 and pkgA-2.i386 to be installed at the same time. However this is very rarely
desired. Install only packages, like the kernel, are exempt from this check.  The default is `1'.
```

If the option is enabled, yum will verify in the transaction building phase in `YumBase.buildTransaction()` that any duplicate packages (as in, having the same `name`) are of the same version. In case they're not, a verbose [message](https://github.com/rpm-software-management/yum/blob/d97a85381215645470604466b30758984f7d7770/yum/__init__.py#L1277) is printed.
